iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1443,6 +1449,25 @@ static void overlayTableProperties(Configuration configuration, TableDesc tableD
     map.remove("columns.comments");
   }
 
+  @Override
+  public void validateCurrentSnapshot(TableDesc tableDesc) {
+    if (conf.getBoolean(ConfigProperties.LOCK_HIVE_ENABLED, TableProperties.HIVE_LOCK_ENABLED_DEFAULT) ||
+        !HiveConf.getBoolVar(conf, ConfVars.HIVE_TXN_EXT_LOCKING_ENABLED)) {
+      return;
+    }
+    Table table = IcebergTableUtil.getTable(conf, tableDesc.getProperties());
```

Comment:
This would be fetching the table from cache, by default `skipCache` is `false`, can that lead to getting an outdated snapshot id in any case?

Reply:
we compare the state of the table in a cache (populated at compile time) with the latest one
````
((BaseTable) table).operations().refresh() 
````